### PR TITLE
Enable connect-engine support

### PR DESCRIPTION
**Pull request by futureweb**
Not quite sure if it's working that way - the first Time I do a MariaDB Spec-File update ... so please review critically ... thx ;-)

### CODE REVIEW

**Comment by futureweb** [comment]
well ... guess it's not working as expected ... no idea why ... :-(
Build was OK ... mariadb103-connect-engine-10.3.21-2.el7.ius.x86_64.rpm can be installed without problem ... but when trying to load the .so ... 

```
2020-01-09 18:12:27 0 [ERROR] mysqld: Can't open shared library '/usr/lib64/mariadb/plugin/ha_connect.so' (errno: 22, undefined symbol: _Z10XMLColumnsP7_globalPcS1_P22ha_table_option_structb)
2020-01-09 18:12:27 0 [ERROR] Couldn't load plugins from 'ha_connect.so'.
2020-01-09 18:12:27 0 [Warning] Plugin 'OQGRAPH' is of maturity level gamma while the server is stable
2020-01-09 18:12:27 0 [ERROR] mysqld: Plugin 'FEDERATED' already installed
2020-01-09 18:12:27 0 [ERROR] mysqld: Can't open shared library '/usr/lib64/mariadb/plugin/ha_connect.so' (errno: 0, undefined symbol: _Z10XMLColumnsP7_globalPcS1_P22ha_table_option_structb)
```

**Comment by futureweb** [comment]
maybe @carlwgeorge got a spare Minute or 2? ;-)

File: mariadb103.spec [hunk 1]
```diff
@@ -1622,6 +1622,9 @@ fi
 %endif
 
 %changelog
+* Thu Jan 09 2020 Andreas Schnederle-Wagner <schnederle@futureweb.at> - 3:10.3.18-2
```
Comment (carlwgeorge) [inline mariadb103.spec]: The version here in the changelog is incorrect, it should be `3:10.3.21-2`.

**Review by carlwgeorge — CHANGES_REQUESTED** [review]
Overall the changes here look fine, just that small adjustment in the changelog entry.

**Comment by carlwgeorge** [comment]
Hey @futureweb, I added a review with one small suggestion, but overall it looks good.  I haven't yet tried to reproduce the error you saw.  Did you build it yourself or did you get it out of the CI system?  If you want to try the latter, open [this build](https://cirrus-ci.com/task/5165460429733888), click the download button next to "results", then extract the zip file which contains all the RPMs from this build.

File: mariadb103.spec (hunk 1 above)
Comment (futureweb) [reply mariadb103.spec]: my bad - overlooked Version, fixed

**Comment by futureweb** [comment]
I already tried the Version out of the CI System ... weird - will give it another shot tomorrow ... guess I made a mistake somewhere ...

**Comment by carlwgeorge** [comment]
Are you all the way up to date on the base operating system?  Errors like `undefined symbol` sometimes are related to an executable linking to a newer version of a library than you have available.  That could be the case if you use an RPM built for CentOS 7.7 (like these are) on an older point release.

**Comment by futureweb** [comment]
I just don't get it why I get this Error ... latest Centos 7.7, (CentOS Linux release 7.7.1908 (Core)) no further updates available ... latest CI Build tested right now ... "rpm --reinstall" all latest CI built RPMs ... same Error again ... *sigh*

`2020-01-30 15:38:31 0 [ERROR] mysqld: Can't open shared library '/usr/lib64/mariadb/plugin/ha_connect.so' (errno: 22, undefined symbol: _Z10XMLColumnsP7_globalPcS1_P22ha_table_option_structb)
2020-01-30 15:38:31 0 [ERROR] Couldn't load plugins from 'ha_connect.so'.
2020-01-30 15:38:31 0 [Warning] Plugin 'OQGRAPH' is of maturity level gamma while the server is stable
2020-01-30 15:38:31 0 [ERROR] mysqld: Plugin 'FEDERATED' already installed
2020-01-30 15:38:31 0 [ERROR] mysqld: Can't open shared library '/usr/lib64/mariadb/plugin/ha_connect.so' (errno: 0, undefined symbol: _Z10XMLColumnsP7_globalPcS1_P22ha_table_option_structb)`

**Review by carlwgeorge — DISMISSED** [review]

**Comment by carlwgeorge** [comment]
The spec file changes look good now, but I don't know what to do about that error.  I was able to reproduce it by trying to load the plugin manually (this normally happens automatically via the `/etc/my.cnf.d/connect.cnf` file).

```
# mysql -e "INSTALL SONAME 'ha_connect';"
ERROR 1126 (HY000) at line 1: Can't open shared library '/usr/lib64/mariadb/plugin/ha_connect.so' (errno: 2, undefined symbol: _Z10XMLColumnsP7_globalPcS1_P22ha_table_option_structb)
```

I'm not able to reproduce the error on Fedora 31 (mariadb-connect-engine-10.3.21-3.fc31.x86_64) or with the upstream MariaDB repos on CentOS 7 (MariaDB-connect-engine-10.3.22-1.el7.centos.x86_64).  I'll need some more time to try to resolve this.

**Comment by carlwgeorge** [comment]
I spoke too soon!  Comparing the Fedora and IUS spec files, I ran across an extra build requirement that was added with 10.3.21.

https://src.fedoraproject.org/rpms/mariadb/c/42251457ba8e232fa353b3292a67214153f99581

More details can be found in the upstream bug report.

https://jira.mariadb.org/browse/MDEV-21450

Add that as part of this pull request with another commit (preferably in the same spot as the Fedora spec file), and I'll squash all the commits together when I merge it.

**Comment by futureweb** [comment]
@carlwgeorge *sigh* ... guess you nailed it (once again) ... ;-)
Commited Change and waiting for CI to finish build ...

**Review by carlwgeorge — APPROVED** [review]

**Comment by carlwgeorge** [comment]
The changes look good, and the CI passed.  Try out those latest RPM artifacts, and if they work for you I'll merge this and get it published to the repos.

**Comment by futureweb** [comment]
>  if they work for you I'll merge this and get it published to the repos.

@carlwgeorge working as expected ... let's go! ;-)

**Comment by carlwgeorge** [comment]
I've published mariadb103-10.3.21-2 to the testing repos.  I'll move them to the main repos next week.

**Comment by carlwgeorge** [comment]
mariadb103-10.3.21-2 has been pushed to the main repos.

**Pull request merged**